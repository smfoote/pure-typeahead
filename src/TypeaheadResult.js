import React, {Component} from 'react'

export default class TypeaheadResult extends Component {
  render() {
    const { children } = this.props;
    return (
      <typeahead-result
        onClick={this.props.onSelect}
        class={this.props.isHighlighted ? 'highlighted' : null}
      >
        {children}
      </typeahead-result>
    );
  }
}
