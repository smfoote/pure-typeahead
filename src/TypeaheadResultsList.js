import React, {Component} from 'react'
import TypeaheadResult from './TypeaheadResult'

export default class TypeaheadResultsList extends Component {

  render() {
    let currentIndex = -1;
    const resultValues = [];
    const children = React.Children.map(this.props.children, child => {
      if (child && child.type === TypeaheadResult) {
        // scope the index to avoid closure problems
        const idx = currentIndex = currentIndex + 1;
        resultValues.push(child.props.value);
        return React.cloneElement(child, {
          isHighlighted: idx === this.props.highlightedIndex,
          _onSelect: (value) => { this.props._onSelect(value); },
        })
      }
      return child;
    });
    this.props.onResultsUpdate(resultValues);
    return <typeahead-results-list class={this.props.className}>
      {children}
    </typeahead-results-list>
  }
}
