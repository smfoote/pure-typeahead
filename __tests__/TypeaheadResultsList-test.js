import React from 'react';
import Enzyme, { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import TypeaheadResultsList from '../src/TypeaheadResultsList';
import TypeaheadResult from '../src/TypeaheadResult';

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    updateHighlightedIndex: jest.fn(),
    onResultsUpdate: jest.fn(),
    highlightedIndex: -1,
  };
  return {
    wrapper: mount(<TypeaheadResultsList {...props}>
      <h3>Cities</h3>
      <TypeaheadResult>Las Vegas</TypeaheadResult>
      <TypeaheadResult>Los Angeles</TypeaheadResult>
      <TypeaheadResult>San Francisco</TypeaheadResult>
    </TypeaheadResultsList>),
    props
  };
};

describe('TypeaheadResultsList', () => {
  it('renders self and subcomponents', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
