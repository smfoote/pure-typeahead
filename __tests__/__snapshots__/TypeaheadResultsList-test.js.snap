// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeaheadResultsList renders self and subcomponents 1`] = `
<TypeaheadResultsList
  highlightedIndex={-1}
  updateHighlightedIndex={[Function]}
>
  <typeahead-results-list>
    <h3
      key=".0"
    >
      Cities
    </h3>
    <TypeaheadResult
      isHighlighted={false}
      key=".1"
    >
      <typeahead-result
        class={null}
        onClick={[Function]}
      >
        Las Vegas
      </typeahead-result>
    </TypeaheadResult>
    <TypeaheadResult
      isHighlighted={false}
      key=".2"
    >
      <typeahead-result
        class={null}
        onClick={[Function]}
      >
        Los Angeles
      </typeahead-result>
    </TypeaheadResult>
    <TypeaheadResult
      isHighlighted={false}
      key=".3"
    >
      <typeahead-result
        class={null}
        onClick={[Function]}
      >
        San Francisco
      </typeahead-result>
    </TypeaheadResult>
  </typeahead-results-list>
</TypeaheadResultsList>
`;
